### ISSUE TITLE
Default IntelliSense doesn't recognize symbols defined in Macros with template and constexpr

### ISSUE THREAD

**Issue by fancl20** [microsoft/vscode-cpptools#1151]
- Operating System and version: macOS 10.13
- VS Code version: 1.17.2
-  C/C++ extension version: 0.14.0

```cpp
#define UNIQ_TYPE(name, ...) using name = UniqType<__VA_ARGS__, hash(#name)>

template <typename T, int n>
class UniqType {};

constexpr int hash(const char *str) {
    return (*str == '\0') ? 5381 : (hash(str + 1) * 33 + *str) % (1 << 25);
}

UNIQ_TYPE(ID, uint32_t);
```

The IntelliSense doesn't recognize the type ID.

**Comment by sean-mcmanus** [comment]
This works for me with 0.14.6. Are you still having an issue?

**Comment by fancl20** [comment]
I tried in 0.14.6 but still not work.
UNIQ_TYPE or UniqType has completion but the type ID declared by the macro can't be completed.

**Comment by sean-mcmanus** [comment]
Okay, I understand now. The hover info works, because it's provided by IntelliSense. The completion (and Go to Definition) doesn't work because the Tag Parser handles those and it doesn't handle macro replacement. These should get fixed after we implement the features using the IntelliSense engine: https://github.com/Microsoft/vscode-cpptools/issues/13 and https://github.com/Microsoft/vscode-cpptools/issues/255 .

**Comment by sean-mcmanus** [comment]
This has been fixed with 0.19.0 -- the ID variable works with completion and Go to Definition.
